cfme/tests/cloud_infra_common/test_tag_objects.py

```diff
@@ -23,31 +24,34 @@ def setup_first_provider():
     providers.setup_a_provider(prov_class="cloud", validate=True, check_existing=True)
 
 
+param_classes = {
+    'Infra Providers': InfraProvider,
+    'Infra VMs': Vm,
+    'Infra Templates': Template,
+    'Infra Hosts': Host,
+    'Infra Clusters': Cluster,
+    'Infra Stores': Datastore,
+
+    'Cloud Providers': CloudProvider,
+    'Cloud Instances': Instance,
+    'Cloud Availabity Zones': AvailabilityZone,
+    'Cloud Flavors': Flavor,
+    'Cloud Tenants': Tenant
+}
+
 pytestmark = [
-    pytest.mark.parametrize("location", [
-        # Infrastructure
-        InfraProvider, Cluster, Host, Datastore, Vm, Template,
-        # Cloud
-        CloudProvider, Instance,
-        "clouds_availability_zones",  # todo: no navmazing for azone
-        "clouds_flavors",  # todo: no navmazing for flavors
-        "clouds_tenants",  # todo: no navmazing for tenants
-        # "clouds_security_groups",  # Does not have grid view selector
-    ]),
+    pytest.mark.parametrize("location", param_classes),
     pytest.mark.usefixtures("setup_first_provider"),
     pytest.mark.tier(3)
 ]
 
 
-def nav_to(location):
-    if isinstance(location, basestring):
-        pytest.sel.force_navigate(location)
-    else:
-        navigate_to(location, 'All')
+def navigate_and_check(location):
```

Comment:
@izapolsk thoughts on this? This is one place where we actually collided, I rolled the any_present call into this function since 2/3 callers were using it, and the other could ignore the return.